src/hooks/useTodos.ts

```diff
@@ -0,0 +1,48 @@
+import { useState } from 'react';
+import { Filter } from '../App';
+import { useDispatch, useSelector } from 'react-redux';
+import { RootState } from '../app/store';
+import { filterSlice } from '../features/filter';
+
+export const useTodos = () => {
+  const dispatch = useDispatch();
+  const [isTodosLoaded, setIsTodosLoaded] = useState(false);
+
+  const [filterType, setFilterType] = useState<Filter>(Filter.All);
+  const [query, setQuery] = useState('');
+
+  // const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  // const [selectedUserId, setSelectedUserId] = useState<number>(0);
```

Comment:
The lines `// const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);` and `// const [selectedUserId, setSelectedUserId] = useState<number>(0);` are commented out. If these states are no longer needed, consider removing them to keep the code clean. Otherwise, ensure they are implemented correctly if required.